### ISSUE TITLE
Websocket access via custom frontend giving error '503 - Endpoint Creation Failed'

### ISSUE THREAD

**Issue by Farhan7786** [tananaev/traccar#2056]
Hi,

I want to use websockets via custom frontend. But it is giving below error:

```
Chrome console:
WebSocket connection to 'ws://url/api/socket' failed: Error during WebSocket handshake: Unexpected response code: 503 

Chrome Network tab:
503
Endpoint Creation Failed
```

This is how i am doing it. Session is created successfully.
```
$.ajax({ 
               url: 'http://url/api/session',
               type: 'POST',
               data: { email: 'someuser', password: 'password'},               
               success: function(){
                   var socket = new WebSocket('ws://url/api/socket');
                    socket.onopen = function () {
                        console.log("onopen");
                    }
                    socket.onclose = function (event) {
                        console.log("onclose");
                    }
                    socket.onmessage = function (event) {
                        console.log("onmessage:" + JSON.stringify(event));
                    }
                    socket.onerror = function (event) {
                        console.log("onerror");
                    }
               }
            });  
```

**Comment by tananaev** [comment]
I had similar problem because my internet provider changes HTTP headers. I would recommend to try HTTPS and see if you get experience same issue.

**Comment by kirantpatil** [comment]
@Farhan7786, any updates on this issue ?

Here is what I did: https://github.com/tananaev/traccar/issues/2547

**Comment by marcovc** [comment]
I have this problem too. Currently I'm trying with 

`curl -i -N -H "Connection: Upgrade" -H "Upgrade: websocket" -H "Host: <HOST>" -u admin:admin http://<HOST>:8082/api/socket`

and getting

`HTTP/1.1 503 Endpoint Creation Failed
Date: Mon, 05 Dec 2016 22:20:10 GMT
Content-Type: text/html; charset=ISO-8859-1
Cache-Control: must-revalidate,no-cache,no-store
Content-Length: 99
Server: Jetty(9.2.19.v20160908)`

**Comment by tananaev** [comment]
What's in the content?

**Comment by marcovc** [comment]
You mean in the reply I get from the server? Nothing, just the error.

**Comment by tananaev** [comment]
That's not true:
```
Content-Length: 99
```

**Comment by marcovc** [comment]
Could it be the size of the error string + headers? I don't see any more output from that command...

**Comment by tananaev** [comment]
No, it clearly says that it's the size of the content, not header(s).

**Comment by marcovc** [comment]
You're right, I had to redirect it to a file to see it. Here it is:

`<!DOCTYPE<html><head><title>Error</title></head><html><body>503 - Service Unavailable</body></html>`

**Comment by tananaev** [comment]
Are you using the latest official release without any modifications? Does it work without proxy?

**Comment by marcovc** [comment]
I'm using version 3.8, using [this docker container](https://github.com/magna-z/docker-traccar/blob/master/Dockerfile).

I'm trying to make it work with a proxy, but for diagnosing the problem the curl command above is going directly to the backend (I mean HOST here is the server where traccar server is running).

**Comment by tananaev** [comment]
Does it work without proxy? If yes, it must be something with your proxy configuration.

**Comment by marcovc** [comment]
Yes, that container works without proxy and the problem is for sure in the proxy configuration. I'm trying to find out what is is exactly. Everything seems to work except the websocket communication. I've seen the info about proxying over SSL, but I'm using haproxy, so the solution does not translate literally.

I'll keep digging.

Thanks!
Marco

**Comment by tananaev** [comment]
Do you have required Apache modules enabled?

**Comment by marcovc** [comment]
I'm using haproxy.

**Comment by tananaev** [comment]
Does it even support WebSockets?

**Comment by marcovc** [comment]
At least according to [this post](http://blog.silverbucket.net/post/31927044856/3-ways-to-configure-haproxy-for-websockets). I'm really no expert in any of this though.

**Comment by tananaev** [comment]
I'm not an expert in HAProxy either. The issue clearly has nothing to do with the Traccar server itself. If you find a solution, please share the info.

**Comment by marcovc** [comment]
Thanks, will do. Marco

**Comment by marcovc** [comment]
Solved it. The problem was that there is cookie being set with "Path=/api" and this was not being rewritten by haproxy. 

Apologies to hijack an issue which apparently is not related to this one.

Thanks for your help
Marco

**Comment by jzahraoui** [comment]
I'm having the same issue
can you help me please.
I post a message here : 
https://www.traccar.org/forums/topic/nginx-alias-configuration/

```
curl -i -N -H "Connection: Upgrade" -H "Upgrade: websocket" -H "Host: 127.0.0.1" -u admin:admin http://127.0.0.1:8082/api/socket
HTTP/1.1 503 Endpoint Creation Failed
Date: Wed, 08 Mar 2017 14:05:03 GMT
Content-Type: text/html; charset=ISO-8859-1
Cache-Control: must-revalidate,no-cache,no-store
Content-Length: 99
Server: Jetty(9.2.21.v20170120)

<!DOCTYPE<html><head><title>Error</title></head><html><body>503 - Service Unavailable</body></html>


```
